src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-announcement-dialog/exam-live-announcement-create-button.component.ts

```diff
@@ -0,0 +1,76 @@
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { faBullhorn } from '@fortawesome/free-solid-svg-icons';
+import dayjs from 'dayjs/esm';
+import { Subscription, from } from 'rxjs';
+
+import { Exam } from 'app/entities/exam.model';
+import { AlertService } from 'app/core/util/alert.service';
+import { ExamLiveAnnouncementCreateModalComponent } from 'app/exam/manage/exams/exam-checklist-component/exam-announcement-dialog/exam-live-announcement-create-modal.component';
+
+@Component({
+    selector: 'jhi-exam-live-announcement-create-button',
+    templateUrl: './exam-live-announcement-create-button.component.html',
+})
+export class ExamLiveAnnouncementCreateButtonComponent implements OnInit, OnDestroy {
+    @Input() exam: Exam;
+
+    faBullhorn = faBullhorn;
+    announcementCreationAllowed = false;
+
+    private modalRef: NgbModalRef | null;
+    private timeoutRef: any;
+    private subscription: Subscription;
+
+    constructor(
+        private modalService: NgbModal,
+        public alertService: AlertService,
+    ) {}
+
+    ngOnInit() {
+        this.checkAnnouncementCreationAllowed();
+    }
+
+    ngOnDestroy() {
+        this.timeoutRef && clearTimeout(this.timeoutRef);
+        this.subscription && this.subscription.unsubscribe();
+    }
+
+    private checkAnnouncementCreationAllowed() {
+        const now = dayjs();
+
+        // Exam must be visible ...
+        const isVisible = !!this.exam.visibleDate?.isBefore(now);
+        // ... and not be completely over, including grace period
+        const endWithGracePeriod = this.exam.endDate?.add(this.exam.gracePeriod || 0, 'seconds');
```

Comment:
The exam end date is the regular end date (without taking time extensions into account, which is done on the server). I think you should still be able to create announcements after the regular working time, aimed just for students with time extensions.

Reply:
There's actually no harm in just letting this work after the exam ended. Announcements will not render for submitted exams.